lib/new-utils.js

```diff
@@ -2,23 +2,34 @@
 
 const Long = require('long');
 
+const maxInt = BigInt(0x100000000);
+const minuOne = BigInt(-1);
 
 /**
  * Converts from bigint provided by napi into Long type used by the datastax library
  * @param {bigint} from
  * @returns {Long}
  */
-function bigint_to_long(from){
-    return Long.fromString(from.toString());
+function bigint_to_long(from) {
+    let lo = from % maxInt;
+    let hi = from / maxInt;
+    if (lo < 0)
+        hi += minuOne
+    return Long.fromValue({ low: Number(lo), high: Number(hi), unsigned: false });
 }
```

Comment:
Looks much better than strings :)

Reply:
That took much longer than it looks like 😅

Reply:
That's perfectly understandable. Diffs rarely reflect the actual amount of work needed to get sth done.